src/main/java/bhs/devilbotz/RobotContainer.java

```diff
@@ -78,10 +78,35 @@ private void buildAutoCommands() {
    * @return the command to run in autonomous
    */
   public Command getAutonomousCommand(AutonomousModes autoMode) {
-    Command autonomousCommand = autoCommands.get(autoMode);
+    Command autonomousCommand = null;
 
     if (autoMode == null) {
       System.out.println("Robot will NOT move during autonomous :/// You screwed something up");
+    } else {
+      switch (autoMode) {
+        case SIT_STILL:
+          break;
+        case MOBILITY:
+          autonomousCommand =
+              new DriveStraightPID(driveTrain, ShuffleboardManager.autoDistance.getDouble(5));
+          break;
+        case SCORE_AND_MOBILITY:
+          break;
+        case DOCK_AND_ENGAGE:
+          break;
+        case MOBILITY_DOCK_AND_ENGAGE:
+          break;
+
+        case SCORE_DOCK_AND_ENGAGE:
+          break;
+
+        case SCORE_MOBILITY_DOCK_ENGAGE:
+          break;
+        case SCORE_MOBILITY_PICK_DOCK_ENGAGE:
+          break;
```

Comment:
you shouldn't need any of this, its a redundant check The autoCommands.get will do all of this for you.

Reply:
(your default case)

Reply:
Hi Parker, the code isn't redundant.  `buildAutoCommands` is initialized in the constructor so those commands can't use any of the user defined values from Shuffleboard. `getAutonomousCommand` is called when the autonomous routine is just about to be executed, so the command can be created on the fly using the Shuffleboard parameters.  So, eventually, other than the "Sit Still" command, the autonomous commands need to be dynamically created.  All of the autonomous routines defined [here](https://github.com/BHSRobotix/ChargedUp2023#desired-autonomous-routines) have at least one configurable parameter.